File: AdjacentElementsProduct/AdjacentElementsProduct.ts
function AdjacentElementsProduct(inputArray : number[]) : number{
    if(inputArray.length == 0) return -1;

    if(inputArray.length == 1) return inputArray[0];

    let largest = inputArray[0] * inputArray[1];

    for(let i = 1; i < inputArray.length - 1; i++){

        const product = inputArray[i] * inputArray[i+1];

        console.log(product);

        largest = largest < product ? product : largest;
    }

    return largest;
}

console.log(AdjacentElementsProduct([11,23]))

File: AdjacentElementsProduct/AdjacentElementsProduct.java
package AdjacentElementsProduct;

public class AdjacentElementsProduct {
    public static int AdjacentElements(int arr[]){
        if(arr.length == 0) return -1;

        if(arr.length == 1) return arr[0];

        int largest = arr[0] * arr[1];

        for(int i = 1; i < arr.length - 1; i++){

            int product = arr[i] * arr[i+1];
    
            largest = largest < product ? product : largest;
        }

        return largest;
    }
    public static void main(String[] args) {

        int arr[] = {11,12,-1,-133};

        System.out.println(AdjacentElements(arr));
    }

}
